hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -450,6 +451,10 @@ runTest :: Test a -> (Either Failure a, [Log])
 runTest =
   runIdentity . runTestT
 
+mapTestT :: Monad m => (b -> m a) -> TestT m b -> TestT m a
+mapTestT f =
+  mkTestT . (=<<) (\(e, l) -> fmap (flip (,) l) $ traverse f e) . runTestT
```

Comment:
I think this new one is actually not that useful, as the `m` can't be changed.

One alternative which gives more power than `hoist` is:
```hs
mapTestT :: (m (Either Failure a, [Log]) -> n (Either Failure b, [Log])) -> TestT m a -> TestT n b
mapTestT f =
  mkTestT . f . runTestT
```

Not sure how useful that one is?